instrumentation/httpasyncclient/src/main/java/brave/httpasyncclient/TracingHttpAsyncClientBuilder.java

```diff
@@ -148,6 +151,37 @@ static void parseTargetAddress(HttpHost target, Span span) {
     }
   }
 
+  final class FutureCallbackWithCurrentSpan<T> implements FutureCallback<T> {
```

Comment:
I'd reduce the cost of this by making it a static final, passing the CurrentTraceContext, TraceContext, and the callback. CurrentTraceContext scope commands are a bit cheaper than tracer based ones.